src/php/Printer/TypePrinter/ArrayPrinter.php

```diff
@@ -4,13 +4,74 @@
 
 namespace Phel\Printer\TypePrinter;
 
+use Phel\Printer\PrinterInterface;
+
+/**
+ * @implements TypePrinterInterface<array>
+ */
 final class ArrayPrinter implements TypePrinterInterface
 {
+    private PrinterInterface $printer;
+    private bool $withColor;
+
+    public function __construct(PrinterInterface $printer, bool $withColor = false)
+    {
+        $this->printer = $printer;
+        $this->withColor = $withColor;
+    }
+
     /**
-     * @param mixed $form
+     * @param array $form
      */
     public function print($form): string
     {
-        return '<PHP-Array>';
+        $arr = $this->isList($form)
+            ? $this->formatValuesFromList($form)
+            : $this->formatKeyValuesFromDict($form);
+
+        return sprintf('<PHP-Array [%s]>', $this->color(implode(', ', $arr)));
+    }
+
+    private function isList(array $form): bool
+    {
+        return array_keys($form) === range(0, count($form) - 1);
+    }
+
+    private function formatValuesFromList(array $form): array
+    {
+        return array_map(
+            fn ($v) => $this->formatValue($v),
+            $form
+        );
+    }
+
+    private function formatKeyValuesFromDict(array $form): array
+    {
+        return array_map(
+            fn ($k, $v) => sprintf('%s:%s', $this->formatValue($k), $this->formatValue($v)),
+            array_keys($form),
+            $form
+        );
+    }
+
+    /**
+     * @param mixed $v
+     */
+    private function formatValue($v): string
+    {
+        if (is_string($v)) {
```

Comment:
Is this required? Can't we just use the String printer?

Reply:
of course we can 🙈 I will change it sofort!